### PR TITLE
Add hostname config option to NGINX plugin

### PR DESCRIPTION
**Pull request by dhrupadb**

### CODE REVIEW

File: src/nginx.c [hunk 1]
```diff
@@ -211,7 +215,11 @@ static void submit (const char *type, const char *inst, long long value)
 
   vl.values = values;
   vl.values_len = 1;
-  sstrncpy (vl.host, hostname_g, sizeof (vl.host));
+  if (hostname != NULL) {
```
Comment (igorpeshansky) [inline src/nginx.c]: Simpler:
```
char *host = hostname != NULL ? hostname : hostname_g;
sstrncpy (vl.host, host, sizeof (vl.host));
```

File: src/nginx.c [hunk 2]
```diff
@@ -49,6 +50,7 @@ static char   nginx_curl_error[CURL_ERROR_SIZE];
 
 static const char *config_keys[] =
 {
+  "Hostname",
```
Comment (igorpeshansky) [inline src/nginx.c]: I initially said that you should also document this in `collectd.conf.pod`, but then realized that the equivalent Apache option is not documented either, so up to you. We could also fix the Apache plugin documentation, unless it's already fixed upstream.

Comment (igorpeshansky) [inline src/nginx.c]: Did we not agree on `Host`, to match Apache's behavior?

**Review by igorpeshansky — CHANGES_REQUESTED** [review]

File: src/nginx.c (hunk 2 above)
Comment (dhrupadb) [inline src/nginx.c]: Done.

File: src/nginx.c (hunk 1 above)
Comment (dhrupadb) [inline src/nginx.c]: Changed. I changed the variable to host as well to keep it consistent with the config option.

File: src/nginx.c [hunk 3]
```diff
@@ -33,6 +33,7 @@
 
 #include <curl/curl.h>
 
+static char *host    = NULL;
```
Comment (qingling128) [inline src/nginx.c]: Should the spacing line up with the variables below?

File: src/nginx.c [hunk 4]
```diff
@@ -97,7 +99,9 @@ static int config_set (char **var, const char *value)
 
 static int config (const char *key, const char *value)
 {
-  if (strcasecmp (key, "url") == 0)
+  if (strcasecmp (key, "host") == 0)
```
Comment (igorpeshansky) [inline src/nginx.c]: There's an opportunity for a refactoring here, but we probably want to make a minimally invasive change, so let's skip that for now.

File: src/nginx.c [hunk 5]
```diff
@@ -211,7 +215,8 @@ static void submit (const char *type, const char *inst, long long value)
 
   vl.values = values;
   vl.values_len = 1;
-  sstrncpy (vl.host, hostname_g, sizeof (vl.host));
+  char *vl_host = host != NULL ? host : hostname_g;
+  sstrncpy (vl.host, vl_host, sizeof (vl.host));
```
Comment (igorpeshansky) [inline src/nginx.c]: We can combine these into one line:
```c
  sstrncpy (vl.host, host != NULL ? host : hostname_g, sizeof (vl.host));
```

**Review by igorpeshansky — CHANGES_REQUESTED** [review]

File: src/nginx.c (hunk 3 above)
Comment (dhrupadb) [inline src/nginx.c]: Fixed

File: src/nginx.c (hunk 4 above)
Comment (dhrupadb) [inline src/nginx.c]: Agreed.

File: src/nginx.c (hunk 5 above)
Comment (dhrupadb) [inline src/nginx.c]: Done.

**Review by dhrupadb — COMMENTED** [review]
PTAL

**Review by igorpeshansky — APPROVED** [review]
LGTM :shipit:

**Comment by qingling128** [comment]
LGTM.

**Comment by igorpeshansky** [comment]
dhrupadb was at Google at the time.

**Pull request merged**